src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2249,6 +2265,30 @@ def update_azure_service_mesh_profile(self) -> ServiceMeshProfile:
             new_profile.istio.certificate_authority.plugin.cert_chain_object_name = cert_chain_object_name
             updated = True
 
+        # deal with mesh upgrade commands
+        if mesh_upgrade_command is not None:
+            requested_revision = self.raw_param.get("revision", None)
+            if mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE or mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_ROLLBACK:
+                if len(new_profile.istio.revisions) < 2:
+                    raise ArgumentUsageError('Azure Service Mesh upgrade is not in progress.')
+                if mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE:
+                    revision_to_remove = min(new_profile.istio.revisions)
+                    revision_to_keep = max(new_profile.istio.revisions)
+                else:
+                    revision_to_remove = max(new_profile.istio.revisions)
+                    revision_to_keep = min(new_profile.istio.revisions)
+                msg = (f"This operation will remove Istio control plane for revision {revision_to_remove}. "
+                       f"Please ensure all data plane workloads have been rolled over to revision {revision_to_keep} so that they are still part of the mesh. "
+                       "\nAre you sure you want to proceed?")
+                if prompt_y_n(msg, default="y"):
+                    new_profile.istio.revisions.remove(revision_to_remove)
+                    updated = True
+            elif mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_START and requested_revision is not None:
+                if new_profile.istio.revisions is None:
```

Comment:
Can / should `revisions` ever be null or empty upon an upgrade start? Do we want to throw an error if it is?

Reply:
Thanks for the catch. Added validation to make sure that mesh is **enabled** on the cluster before `aks mesh upgrade` commands can be called. This will ensure that `new_profile.istio.revisions` is never None. In any case if this happens, I would like this to be handled by rp since something same can happen from the ARM path as well. Let me know your thoughts.

Reply:
Think we should be good. Based on discussion on FSM PR with @SanyaKochhar, looks like we have this covered from an RP level to ensure that revisions is only empty for fresh installations.